### PR TITLE
Updated API docs

### LINKED ISSUE #56: Incomplete Documentation 1/3 (API)
Current [docs](https://membrane-curvature.readthedocs.io/en/latest/) are incomplete and lack API docs. 

To fix this issue:
- [ ] Add API docs using reStructured Text, docstrings and autosummary. (Include `MembraneCurvature` class and functions  in  `curvature.py` and `surface.py`)
- [ ] Update `index.rst`
- [ ] Update `getting_started.rst`

@lilyminium hopefully I am not missing anything here at the moment. :)

### PR DESCRIPTION
**Pull request by ojeda-e**
This is a preliminary version of updated API docs. It may help to fix #56 

This PR includes:
- [x] Updated API docs
- [x] Updated `index.rst`  

`getting_started.rst` is not really ready.

But maybe I can have some review for what I have so far @lilyminium?
Any suggestions are appreciated @orbeckst @IAlibay @fiona-naughton

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,36 +3,57 @@
...
+------- 
+
+Source code included in this project is available under the `GNU Public
+Licence v3`_ from `github.com/MDAnalysis/membrane_curvature`_.
+
+
+.. _GNU Public Licence v3:
+   http://www.gnu.org/licenses/old-licenses/gpl-3.0.html
+.. _MDAnalysis: https://www.mdanalysis.org
+.. _`github.com/MDAnalysis/membrane_curvature`: http://github.com/MDAnalysis/membrane_curvature
```
Comment (ojeda-e) [inline docs/index.rst:56]: Broken link. It should be http://github.com/MDAnalysis/membrane-curvature

Comment (orbeckst) [reply docs/index.rst:56]: Actually, use SSL whenever possible: https

File: membrane_curvature/base.py [hunk 2]
```diff
@@ -98,17 +98,22 @@ class MembraneCurvature(AnalysisBase):
     The respective 2D curvature plots can be obtained using the `matplotlib` package for
     data visualization via `imshow`. We recommend using the `gaussian` interpolation. 
 
-    A simple plot using `imshow` can be obtained by
+    A simple plot using `imshow` can be obtained by::
```
Comment (orbeckst) [inline membrane_curvature/base.py:101]: Maybe remove this text from the Python doc string and move it to a dedicated reST file where you
1. show a real example with data
2. include actual images

That makes the example much more impactful.

**Review by orbeckst — APPROVED** [review]
Looking already good, see comments.

**Comment by ojeda-e** [comment]
Thanks for your comments @orbeckst. I added the requested changes. API docs are now split into 3 pages.

**Pull request merged**